site/en/docs/extensions/mv3/webhid/index.md

```diff
@@ -0,0 +1,45 @@
+---
+layout: "layouts/doc-post.njk"
+title: "WebHID in extensions"
+seoTitle: "Chrome Extensions: WebHID"
+date: 2023-11-02
+description: A Human Interface Device (HID), which takes input from or provides output to humans, is available in extensions.
+---
+
+A Human Interface Device (HID) takes input from or provides output to humans. It also refers to the HID protocol, a standard for bi-directional communication between a host and a device that is designed to simplify the installation procedure.
+
+This page describes aspects of the API that are particular to extensions. Refer to MDN for complete details of the [WebHID API](https://developer.mozilla.org/docs/Web/API/WebHID_API).
+
+You can find a [sample app for WebHID](https://github.com/GoogleChrome/chrome-extensions-samples/tree/main/functional-samples/sample.co2meter) in our samples repo.
+
+## Availability in extensions
+
+Chrome 117 or later.
+
+## Permissions
+
+No manifest file permissions are required; however WebHID triggers the browser's user permission flow.
+
+## Manifest
+
+No manifest keys are needed for this API.
+
+## Supporting contexts
+
+This API may be used in any extension component. Although this API cannot be used in web service workers, most of its methods and properties can be used in extension service workers. See the next section for details.
+
+## Chrome extension differences
+
+Although WebHID is available to extension service workers, [`WebHID.requestDevice()`](https://developer.mozilla.org/docs/Web/API/HID/requestDevice), which returns a promise that resolves with an [HIDDevice](https://developer.mozilla.org/docs/Web/API/HIDDevice) instance, cannot be called in an extension service worker. To get around this, call `requestDevice()` from an extension page other than the extension service worker and pass the reference to the extension service worker.
+
+The following code follows a typical pattern by calling `requestDevice()` as part of a permissions flow requiring a user gesture. (If the permission has not already been granted, calling `requestDevice()` triggers it.)
+
+```javascript
+someButton.addEventListener('click', async () => {
+  const device = await navigator.usb.requestDevice({
+    filters: [{ vendorId: 1241, productId: 41042 }]
+  });
+  console.log('Device access granted!', device);
+  chrome.runtime.sendMessage(device);
```

Comment:
I'd recommend showing both page and service worker code to make it clear.

```js
// popup.js
myButton.addEventListener("click", async () => {
  const device = await navigator.usb.requestDevice({
    filters: [{ vendorId: 0x1234, productId: 0x5678 }],
  });
  chrome.runtime.sendMessage("newDevice");
});

```
  
```js
// serviceworker.js
chrome.runtime.onMessage.addListener(async (message) => {
  if (newMessage === "newDevice") {
    const devices = await navigator.hid.getDevices();
    for (const device of devices) {
      // open device connection.
      await this.device.open();
    }
  }
});
```

Reply:
Ah, interesting. I was under the impression the extension service worker needed to be post'ed the `device`, but of course this can't work, as the device isn't serializable now that I think about it. So _very_ good point to show both. 

Careful, @beaufortfrancois: your code above mixes HID and USB, though.

Reply:
Ah, the `const device` seems superfluous then.

Reply:
Thanks @tomayac. I've edited code snippet

Reply:
What about ` const device = await navigator.hid.requestDevice`? You don't need to store the reference in a variable, no?

Reply:
This variable is not required indeed.

Reply:
I just pushed an update. PTAL